rumqttd/src/router/routing.rs

```diff
@@ -48,8 +50,12 @@ pub enum RouterError {
     InvalidFilterPrefix(Filter),
     #[error("Invalid client_id {0}")]
     InvalidClientId(String),
+    #[error("Disconnection (Reason: {0:?})")]
+    Disconnect(DisconnectReasonCode),
```

Comment:
There is already a `Disconnected` variant here, cannot we reuse that?

Reply:
Disconnect with Reason code only for v5. Working on Disconnection related stuff in depth is beyond the scope of this PR. So I kept both the variants, it might change in future :)